src/anyp/Uri.cc

```diff
@@ -370,6 +371,8 @@ AnyP::Uri::parse(const HttpRequestMethod& method, const SBuf &rawUrl)
                 rfc1738_unescape(login);
             }
 
+            auto validateAsIpv6 = false;
+
             /* Is there any host information? (we should eventually parse it above) */
             if (*foundHost == '[') {
```

Comment:
Detection of a `[` at the start of `foundHost` should immediately reject in IPv4-only. That character is an IPv6 specification feature - even if the content is `IPvFuture`.

Reply:
"Immediately" is debatable -- refactored parsing code may benefit from postponing checking for that (exceptional) "IPv4-only" condition -- but let's not debate that until this code is actually refactored to use parseHost(), as planned. We want to minimize changes in _this_ PR, and that concern may become moot at that point.

Reply:
My request is to further minimize from what is done already. Replacing the `validateAsIpv6` hack by just throwing on the additional IPv6-disabled criteria.
```suggestion
            if (*foundHost == '[') {
                if (!Ip::EnableIpv6)
                    throw TextException("unsupported IPv6-like address in uri-host", Here())
```

Reply:
I am strongly against this requested change because it duplicates an important decision -- whether to reject an expected IPv6 addresses in non-CONNECT URIs when Ip::EnableIpv6 is falsy. That and similar IPv6-related decisions (and the corresponding throwing code) should be done in one place -- where validateAsIpv6 is processed in the current PR code. Spreading that code around various parsing `if`s makes the code significantly worse overall, especially when the structure/state of those `if`s is already quite complex and expected to change.

This particular code snippet is _not_ about parsing or interpreting an address (as IPv6). It is about isolating a [bracketed] address. Please keep address parsing/interpretation decisions confined to one place, after the address is isolated by the existing address isolation code. Avoid mixing those two concerns together.

The same objection applies to the other similar/dependent change requests, of course.